emcc.py

```diff
@@ -2738,9 +2759,10 @@ def modularize():
     f.write(src)
 
     # Export using a UMD style export, or ES6 exports if selected
+
     if shared.Settings.EXPORT_ES6:
       f.write('''export default %s;''' % shared.Settings.EXPORT_NAME)
-    else:
+    elif shared.Settings.target_environment_may_be('node') or shared.Settings.target_environment_may_be('shell'):
```

Comment:
why is this only for node and shell?

Reply:
Oh, I thought RequireJS/AMD modules were specific to shells, but looking closer, they are indeed used in browsers.

This makes me shake my head how ever did we let this code to land.. not many people use AMD modules, but we unconditionally generate the code in all builds. Not cool. The EXPORT_ES6 support above is well thought out.

Options here may be to put this behind a `-s EXPORT_AMD=1` flag (or `-s EXPORT_REQUIREJS=1` flag), perhaps default on if we don't want to break, and users can opt out with `-s EXPORT_AMD=0` if they do not care about it.

`MINIMAL_RUNTIME` would default to `EXPORT_AMD=0`.

(Or alternatively this feature would be disabled altogether in `MINIMAL_RUNTIME`, although that is perhaps unnecessarily diverging)

Reply:
Possibly the ES6 path could be made the default, since that's the clear future.

Reply:
`EXPORT_METHOD` with `none`, `umd`, and `esm/es6` options would probably be best.

Reply:
What did we decide here? The `node,shell` code here doesn't look right.

Reply:
I'd be ok to say that MINIMAL_RUNTIME only supports ES6 export.

Reply:
check